gusto/common_forms.py

```diff
@@ -194,3 +198,65 @@ def diffusion_form(test, q, kappa):
     form = -inner(test, div(kappa*grad(q)))*dx
 
     return diffusion(form)
+
+
+def split_continuity_form(equation):
+    u"""
+    Loops through terms in a given equation, and splits continuity terms into
+    advective and divergence terms.
+
+    This describes splitting ∇.(u*q) into u.∇q and q(∇.u),
+    for transporting velocity u and transported q.
+
+    Args:
+        equation (:class:`PrognosticEquation`): the model's equation.
+
+    Returns:
+        equation (:class:`PrognosticEquation`): the model's equation.
+    """
+
+    for t in equation.residual:
+        if (t.get(transport) == TransportEquationType.conservative):
+            # Get fields and test functions
+            subj = t.get(subject)
+            prognostic_field_name = t.get(prognostic)
+            if hasattr(equation, "field_names"):
+                idx = equation.field_names.index(prognostic_field_name)
+            else:
+                idx = equation.field_name.index(prognostic_field_name)
```

Comment:
I think this won't give what we expect! Is `field_name` a string and not a list?

In the case that we don't have `field_names`, we don't want an index and don't want to split anything up (because we won't have a mixed function space)